core/src/test/java/io/grpc/internal/ClientCallImplTest.java

```diff
@@ -878,6 +880,34 @@ public void getAttributes() {
     assertEquals(attrs, call.getAttributes());
   }
 
+  @Test
+  public void sendMessage_closeMarshallerInputStream() throws Exception {
+    final InputStream mockInputStream = mock(InputStream.class);
+    Marshaller<Void> marshaller = new Marshaller<Void>() {
+      @Override
+      public InputStream stream(Void value) {
+        return mockInputStream;
+      }
+
+      @Override
+      public Void parse(InputStream stream) {
+        throw new UnsupportedOperationException();
+      }
+    };
+    MethodDescriptor<Void, Void> method = MethodDescriptor.<Void, Void>newBuilder()
```

Comment:
TestMethodDescriptors.voidDescriptor()

Reply:
In this case I  need to wire up the mockInputStream to the methodDescriptor, so I can't use the canned MethodDescriptor.

Reply:
TestMethodDescriptors.voidDescriptor().toBuilder().setRequestMarshaller().build()